src/__smoke-test__/smoke-test.spec.ts

```diff
@@ -0,0 +1,446 @@
+            filterSuccessfulResponse(response, requestBody)
+        );
+
+        // FID and LCP require user interaction and clear does not trigger other events
+        const clearButton = page.locator('[id=clearRequestResponse]');
+        await clearButton.click();
+
+        // Wait until fetch is complete
+        await page.waitForTimeout(10000);
+
+        // Parse payload
+        const parsedPayload = parseBatches(requestBody);
+        const lcp = getEventsByType(parsedPayload, LCP_EVENT_TYPE);
+        const fid = getEventsByType(parsedPayload, FID_EVENT_TYPE);
+
+        // Web vital events are only emitted once
+        expect(lcp.length).toEqual(1);
+        expect(fid.length).toEqual(1);
+
+        // Parse metadata to verify pageId
+        const lcpPageId = JSON.parse(lcp[0].metadata).pageId;
+        const fidPageId = JSON.parse(fid[0].metadata).pageId;
+        expect(lcpPageId).toEqual(expectedPage);
+        expect(fidPageId).toEqual(expectedPage);
+
+        // Add all event ids to eventIds array
+        addEventsToList(parsedPayload, eventIds);
+    });
+
+    test('test navigation events using pushState are emitted', async () => {
+        const PAGEVIEW_COUNT = 2;
+        const NAVIGATION_COUNT = 2;
+        const expectedPages = ['/page_view_one', '/page_view_two'];
+        const requestBody = [];
+
+        // Selector for pushStateOne
+        const pushStateOne = page.locator('[id=pushStateOneToHistory]');
+        const pushStateTwo = page.locator('[id=pushStateTwoToHistory]');
+
+        // Trigger pushStateOne
+        await page.waitForTimeout(300);
+        await pushStateOne.click();
+
+        // Collect successful requests
+        page.on('response', (response) =>
+            filterSuccessfulResponse(response, requestBody)
+        );
+
+        // Wait until fetch is complete
+        await page.waitForTimeout(5000);
+
+        // Trigger pushStateTwo
+        await pushStateTwo.click();
+
+        // Wait until fetch is complete
+        await page.waitForTimeout(5000);
+
+        // Parse payload
+        const parsedPayload = parseBatches(requestBody);
+        const navigation = getEventsByType(
+            parsedPayload,
+            PERFORMANCE_NAVIGATION_EVENT_TYPE
+        );
+        const pageView = getEventsByType(parsedPayload, PAGE_VIEW_EVENT_TYPE);
+
+        // Verify action is captured
+        expect(pageView.length).toEqual(PAGEVIEW_COUNT);
+        expect(navigation.length).toEqual(NAVIGATION_COUNT);
+
+        // Parse metadata to verify pageId
+        for (let i = 0; i < pageView.length; i++) {
+            const pageId = JSON.parse(pageView[i].metadata).pageId;
+            const navigationPageId = JSON.parse(pageView[i].metadata).pageId;
+            expect(pageId).toEqual(expectedPages[i]);
+            expect(navigationPageId).toEqual(pageId);
+        }
+
+        // Add all event ids to eventIds array
+        addEventsToList(parsedPayload, eventIds);
```

Comment:
issue: What behavior does this test specify? And what property does it need to verify. 

I.e., if we name the test _"When x then y"_, then what is _y_?

In general, we want tests to specify a single behavior.